src/3d/chunks/qgschunkedentity_p.cpp

```diff
@@ -739,6 +748,12 @@ void QgsChunkedEntity::cancelActiveJob( QgsChunkQueueJob *job )
   job->deleteLater();
 }
 
+void QgsChunkedEntity::stopAllJobs()
+{
+  mIsValid = false;
```

Comment:
Do we really need to add `mIsValid` ? It makes the logic more complicated, and code in various places should account for it... It is also not clearly defined what it does - if mIsValid==false, shall we just stop loading nodes? or also stop updating the entity when scene changes?

If we really need an extra flag, I would suggest to have something more explcit - e.g. `mStartJobsAllowed` - that would determine whether startJobs() is allowed to start any new jobs or not. So when the terrain gets deleted, 3D map scene would first call cancelActiveJobs() followed by setStartJobsAllowed(false). This may be useful also in other scenarios - for example when we reach entity's memory limit, we would not start any new jobs until some memory gets freed.

Reply:
This PR does not introduce the `mIsValid` flag. It was introduced by this [commit](https://github.com/qgis/qGIS/commit/3b86c51b25b311aa06d2509339b54bb6b00e47fb). I'm not exactly sure what it supposed to achieve. It looks like, it prevents generating the terrain if the underlying layer is not valid.

Reply:
Ah right, sorry :-) It looks like it is meant to prevent any updates of the scene when the source of the layer is invalid...